doc/architectural/goto-harness.md

```diff
@@ -63,34 +80,66 @@ The options we pass to `goto-harness` are:
 * `harness-type` is the type of harness to generate (`call-function` is the
   function-call harness generator)
 * `function` is the name of the function that gets instrumented
-* then we pass the input GOTO-binary and a name for the output GOTO binary.
+* then we pass the input GOTO-binary and a name for the output C file.
 
-What comes of this is a GOTO binary roughly corresponding to the following C
-pseudocode:
+What comes out of this is a C file that looks like this:
 
 ```C
-// modified_goto_binary.c
-
-#include <assert.h>
-
-int function_to_test(int some_argument)
+// function_to_test
+// file main.c line 3
+signed int function_to_test(signed int some_argument);
+// harness
+// 
+void harness(void);
+// type_constructor_int
+// 
+void type_constructor_int(signed int depth_int, signed int *result_int);
+
+// __GOTO_HARNESS::max_depth
+// file __GOTO_HARNESSharness.c
+signed int max_depth=2;
+// __GOTO_HARNESS::min_depth
+// file __GOTO_HARNESSharness.c
+signed int min_depth=1;
+
+// harness
+// 
+void harness(void)
 {
-  assert(some_argument == 0);
-  return some_argument;
+  signed int some_argument;
+  type_constructor_int(0, &some_argument);
+  function_to_test(some_argument);
 }
 
-void harness(void)
+// type_constructor_int
+// 
+void type_constructor_int(signed int depth_int, signed int *result_int)
 {
-  int argument = nondet_int();
-  function_to_test(argument);
+  signed int nondet;
+  *result_int = nondet;
 }
 ```
 
-You can pass that modified GOTO binary to `cbmc` and select `harness` as the
-entry function for analysis, or further modify it with `goto-harness` or
-`goto-instrument` or other tools.
+After you have generated the harness file, you have two choices. The first
+one is to pass it along with the original file containing the function to
+be instrumented to CBMC for analysis, something that would look like this:
+
+```sh
+$ cbmc --function harness harness.c main.c
+[...]
+
+[function_to_test.assertion.1] line 5 assertion some_argument == 0: FAILURE
+
+** 1 of 1 failed (2 iterations)
+VERIFICATION FAILED
+```
+
+Or, you could choose to turn it into another goto binary with `goto-cc`, to
+be served as input to another instrumentation tool (like `goto-instrument`).
 
-The example above demonstrates the most simple case, which is roughly the same
+---
+
+The example above demonstrates the simplest case, which is roughly the same
 as the entry point `cbmc` automatically generates for functions. However, the
 `function-call` harness can also non-deterministically initialise global
```

Comment:
⛏️ 
```suggestion
`function-call` harness can also non-deterministically initialize global
```

Reply:
I'd much rather not, especially since this bit hasn't been touched by this PR.

Unless you object to that strictly, of course.